web/server.js

```diff
@@ -158,9 +159,28 @@ app.get('/', (req, res) => {
     }
     const { locale } = res;
     const fullUrl = 'https://www.electricitymap.org' + req.originalUrl;
+
+    // basic auth and time machine cookie if we're in the back in time container
+    if (process.env['BASIC_AUTH_USERNAME;']){
+      let user = auth(req);
+      let authorized = false;
+      if (user){
+        let ix = process.env['BASIC_AUTH_USERNAMES'].split(';').indexOf(user.name);
+        if (ix !== -1 && user.pass === process.env['BASIC_AUTH_PASSWORDS'].split(';')[ix]){
```

Comment:
It's a tad dangerous because this forces us to have two lists (one for users, one for passwords), and is therefore prone to errors (if you offset one item for example).
A better way would be to serialise auth using `user1:pass1,user2:pass2`...